cli/packages/cmd/run.go

```diff
@@ -237,80 +345,95 @@ func filterReservedEnvVars(env map[string]models.SingleEnvironmentVariable) {
-func executeSingleCommandWithEnvs(args []string, secretsCount int, env []string) error {
-	command := args[0]
-	argsForCommand := args[1:]
-
-	log.Info().Msgf(color.GreenString("Injecting %v Infisical secrets into your application process", secretsCount))
+func createInjectableEnvironment(request models.GetAllSecretsParameters, projectConfigDir string, secretOverriding bool, shouldExpandSecrets bool, token *models.TokenDetails) (models.InjectableEnvironmentResult, error) {
 
-	cmd := exec.Command(command, argsForCommand...)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Env = env
+	secrets, err := util.GetAllEnvironmentVariables(request, projectConfigDir)
 
-	return execCmd(cmd)
-}
+	if err != nil {
+		return models.InjectableEnvironmentResult{}, err
+	}
 
-func executeMultipleCommandWithEnvs(fullCommand string, secretsCount int, env []string) error {
-	shell := [2]string{"sh", "-c"}
-	if runtime.GOOS == "windows" {
-		shell = [2]string{"cmd", "/C"}
+	if secretOverriding {
+		secrets = util.OverrideSecrets(secrets, util.SECRET_TYPE_PERSONAL)
 	} else {
-		currentShell := os.Getenv("SHELL")
-		if currentShell != "" {
-			shell[0] = currentShell
+		secrets = util.OverrideSecrets(secrets, util.SECRET_TYPE_SHARED)
+	}
+
+	if shouldExpandSecrets {
+
+		authParams := models.ExpandSecretsAuthentication{}
+
+		if token != nil && token.Type == util.SERVICE_TOKEN_IDENTIFIER {
+			authParams.InfisicalToken = token.Token
+		} else if token != nil && token.Type == util.UNIVERSAL_AUTH_TOKEN_IDENTIFIER {
+			authParams.UniversalAuthAccessToken = token.Token
 		}
+
+		secrets = util.ExpandSecrets(secrets, authParams, projectConfigDir)
 	}
 
-	cmd := exec.Command(shell[0], shell[1], fullCommand)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Env = env
+	secretsByKey := getSecretsByKeys(secrets)
+	environmentVariables := make(map[string]string)
 
-	log.Info().Msgf(color.GreenString("Injecting %v Infisical secrets into your application process", secretsCount))
-	log.Debug().Msgf("executing command: %s %s %s \n", shell[0], shell[1], fullCommand)
+	// add all existing environment vars
+	for _, s := range os.Environ() {
+		kv := strings.SplitN(s, "=", 2)
+		key := kv[0]
+		value := kv[1]
+		environmentVariables[key] = value
+	}
 
-	return execCmd(cmd)
-}
+	// check to see if there are any reserved key words in secrets to inject
+	filterReservedEnvVars(secretsByKey)
 
-// Credit: inspired by AWS Valut
-func execCmd(cmd *exec.Cmd) error {
-	sigChannel := make(chan os.Signal, 1)
-	signal.Notify(sigChannel)
+	// now add infisical secrets
+	for k, v := range secretsByKey {
+		environmentVariables[k] = v.Value
+	}
 
-	if err := cmd.Start(); err != nil {
-		return err
+	// Create and sort the env slice using slices.SortFunc
+	env := make([]string, 0, len(environmentVariables))
+	for key, value := range environmentVariables {
+		env = append(env, key+"="+value)
 	}
```

Comment:
wait I don't see the sort anywhere?

Reply:
Sorry, leftover comment. We only need to sort the secrets, because if we don't sort it, we can't be sure that the ETag is reproducible